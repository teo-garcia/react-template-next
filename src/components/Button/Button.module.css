.button {
  border: 0;
  cursor: pointer;
}