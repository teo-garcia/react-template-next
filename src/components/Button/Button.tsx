import styles from "./Button.module.css";
import classNames from "classnames";
import { ButtonProps } from "../../types/client";

function Button(props: ButtonProps) {
  const { className } = props;
  return <button className={classNames(styles.button, className)} {...props} />;
}

export { Button };
