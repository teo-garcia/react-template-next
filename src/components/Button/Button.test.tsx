import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Button } from './Button';

describe('<Button /> tests', function () {
  test('Should match with the component snapshot', function () {
    const { asFragment } = render(<Button>Hello World</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should render the provided children prop', function () {
    render(<Button>Hello World</Button>);
    expect(screen.getByText(/Hello World/i)).toBeInTheDocument();
  });

  test('Should receive event handlers', function () {
    const mockEventHandler = jest.fn();
    render(<Button onClick={mockEventHandler}>Hello World</Button>);
    userEvent.click(screen.getByText(/hello world/i));
    expect(mockEventHandler).toHaveBeenCalled();
  });
});
