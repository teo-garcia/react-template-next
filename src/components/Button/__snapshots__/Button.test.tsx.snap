// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Hello World
  </button>
</DocumentFragment>
`;
